### ISSUE TITLE
Difficulty displaying BingMap through Angular Route

### ISSUE THREAD

**Issue by charldt81** [infusion-code/angular-maps#78]
Hi There,

I'm in need of a bit of help...

I am trying to use the following example of yours 'Bing Map with Font Based Marker (use any icon font)' and I do get it to display properly when I do a new Angular project, but if a integrate it into an existing project of mine using a 'Route' directive, I do not seem to get the Map to display upon load. I get the box size of the map on the screen, but no map and no errors being displayed.

My 'map.component.ts', 'map.component.html', 'map.component.scss' and 'map.route.ts' is nested in a directory called 'entities' and the 'entities' directory is nested in my 'app' directory.

I have a 'entity.route.ts' which imports the 'map.route.ts' and then exports it to 'entity.module.ts' which is then imported into my ' app.module.ts'.

I have the same code in my 'app.module.ts' as you have, and also the same code as you have in my 'map.component.ts', 'map.component.html', 'map.component.scss' files.

Your 'index.html' link for font-awesome I have placed in the 'head' tags of my 'index.html' file, and in the 'body' of my 'index.html' file, I call my open and closing 'main' '/main' tag directives where I have my opening and closing 'router-outlet' '/router-outlet' tags in my 'main.component.html' file which receives the BingMap but does not render it.

I don't know if I've missed something, somewhere... would you be able to please help me with this issue?

Thank you

**Comment by thor-schueler** [comment]
When you debug, do you hit any of the map services? It sounds almost like the providers are not correctly registered in the module in which you are trying to use the map. Any chance you can create a skeleton that reflects your project setup and post it for us to take a look? If you try to use the map in your main module (temporarily, just to see) does it work?

There should be no problem with using the map in modules, we have done that before plenty of times. So I do suspect it is just the plumbing...

**Comment by charldt81** [comment]
Thank you so much for your response, it actually helped me a lot.
I did in fact mess up my 'modules', I had everything set up in my
'app.module' instead of my 'entities.module', as soon as I transferred
everything over to the correct module it worked perfectly.

I am looking to use this map as a live data map whereby data can pull from
a server and displayed as soon as someone/client logs into this
application, this would be possible to do right?
I'm thinking to use GeoJson hooks in a database and call that to be
displayed on the map upon map request?
I hope that makes sense as I am fairly new to programming and Angular and
I'm learning as I go along.

Thank you again so much for your reply.
